server/php-query-data.cpp

```diff
@@ -4,112 +4,27 @@
 
 #include "server/php-query-data.h"
 
-#include <cassert>
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
-#include <unistd.h>
-
-#include "common/dl-utils-lite.h"
-
-static void *memdup(const void *src, size_t x) {
-  if (!src) {
-    assert(!x);
-    return nullptr;
-  }
-  void *res = malloc(x);
-  assert(res);
-  memcpy(res, src, x);
-  return res;
-}
+namespace {
+  constexpr int max_storage_size = std::max(std::max(sizeof(http_query_data), sizeof(rpc_query_data)), sizeof(job_query_data));
+  constexpr int max_storage_align = std::max(std::max(alignof( http_query_data), alignof(rpc_query_data)), alignof(job_query_data));
+  std::aligned_storage_t<max_storage_size, max_storage_align> php_query_data_storage;
```

Comment:
aligned_storage_t [is deprecated](https://en.cppreference.com/w/cpp/types/aligned_storage)